In [9]:
#Load the data
library(tidyverse)

players <- read_csv("players.csv")
summary(players)
sessions <- read_csv("sessions.csv")
summary(sessions)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

**Data Description**


**Players:**
- There are a total of 196 observations which include 7 different variables
- Summary statistics:
|  Variable  |  Minimum   | 1st Quartile | 3rd Quartile |   Median   |    Mean    |  Maximum   |    NA's    |
| ---------- | ---------- |  ----------  |  ----------  | ---------- | ---------- | ---------- | ---------- |
| played_hours | 0.00 | 0.00 | 0.60 | 0.10 | 5.85 | 223.10 | 0
| Age | 9.00 | 17.00 | 22.75 | 19.00 | 21.14 | 58.00 | 2

|  Variable  |    Type    |  Meaning   |
| ---------- | ---------- | ---------- |
| experience | Character  | Gaming experience <br> (beginner, amateur, pro, veteran) |
|  subscribe |   Logical  | Tells us if player is subscribed to newsletter|
| hashedEmail | Character | Anonymous unique user id used to link players to their sessions |
| played_hours | Numeric | Total number of hours the player has spent on the server |
| name | Character | First name of player |
| gender | Character | Gender of player |
| Age | Numeric | Age of player |

Issues:
- Age variable has two missing values
- Most players recorded almost no playtime while a few accumulated very high totals
     - This may influence model performance
- All demographic information is self-reported, so players may provide inaccurate ages or experience levels
- The response variable subscribe has many more TRUE than FALSE values
- hashedEmail is anonymized, duplicate players cannot be manually verified.


**Sessions:**
- There are a total of 1,535 observations which include 5 different varibales
- Summary statistics:
|  Variable  |  Minimum   | 1st Quartile | 3rd Quartile |   Median   |    Mean    |  Maximum   |    NA's    |
| ---------- | ---------- |  ----------  |  ----------  | ---------- | ---------- | ---------- | ---------- |
| original_start_time | 1.72e12 | 1.72e12 | 1.72e12 | 1.72e12 | 1.72e12 | 1.73e12 | 0
| original_end_time | 1.71e12 | 1.72e12 | 1.72e12 | 1.72e12 | 1.72e12 | 1.73e12 | 2

|  Variable  |    Type    |  Meaning   |
| ---------- | ---------- | ---------- |
| hashedEmail | Character  | Anonymous unique user id used to link players to their sessions |
|  start_time | Character | Start time of session |
| end_time | Character | End time of session |
| original_start_time | Numeric | Unix timestamp stating start time of session |
|original_end_time | Numeric | Unix timestamp stating the end time of session |

Issues:
- The start_time and end_time variables are stored as character strings rather than proper datetime formats
- The numeric timestamps (original_start_time and original_end_time) include two missing values
- Some sessions may be extremely short due to players logging in accidentally or disconnecting, which introduces noise/untidy data
      

**Question**

Broad Question: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

Specific: Can players experience and total hours played predict whether a player subrcribes to the newsletter?


**Exploratory Data Analysis and Visualization**

**Methods and Plan**

**GitHub Repository**
